public/static/js/spawned_hosts.js

```diff
@@ -221,6 +224,28 @@ mciModule.controller('SpawnedHostsCtrl', ['$scope', '$window', '$timeout', '$q',
       );
     };
 
+    $scope.addNewVolumes = function () {
+        mciSpawnRestService.getVolumes(
+            {
+                success: function (resp) {
+                    var volumes = resp.data;
+                    _.each(volumes, function (volume) {
+                        var foundVolume = false;
+                        for (var i = 0; i < $scope.volumes.length; i++) {
```

Comment:
the [some function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) might be good for this.